### PR TITLE
Add test for testing memory usage for celery task

### PR DESCRIPTION
**Pull request by 8r2y5**
I want to merge this change because this test will check memory usage of that task

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

**Review by fowczarek — COMMENTED** [review]
Should we also register `slow` mark in PyTest confing in `saleor/setup.cfg`?

**Comment by NyanKiyoshi** [comment]
> Should we also register `slow` mark in PyTest confing in `saleor/setup.cfg`?

Not needed in this case as there is a fixture defining its existence:

https://github.com/saleor/saleor/blob/e0ff00724d601bb975edfc1cb83797f57bc30f98/conftest.py#L37-L38

e.g. we have this fixture in Saleor tests but not defined explicitly in our `setup.cfg` https://github.com/NyanKiyoshi/pytest-django-queries/blob/2aab6237ad6fee2ef836a7c34927a5876bd58513/pytest_django_queries/plugin.py#L107-L111

Whenever `pytest` starts, it will load the configuration fixtures to allow them to alter the behavior of pytest.

File: conftest.py [hunk 1]
```diff
@@ -24,6 +24,33 @@
     "saleor.tax.tests.fixtures",
 ]
 
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--run-slow",
+        action="store_true",
+        default=False,
+        help="run tests marked as slow",
```
Comment (maarcingebala) [inline conftest.py:33]:
```suggestion
        help="Run tests marked as slow.",
```

File: saleor/tests/fixtures.py [hunk 2]
```diff
@@ -7174,3 +7175,38 @@ def transaction_session_response():
...
+        chunk = tuple(itertools.islice(it, size))
+        while chunk:
+            yield chunk
+            chunk = tuple(itertools.islice(it, size))
+
+    variants_per_product = 4
+    products_count = 10000
+    slug_generator = (f"test-slug-{i}" for i in range(products_count))
+
+    with transaction.atomic():
```
Comment (maarcingebala) [inline saleor/tests/fixtures.py:7193]: Does it have to be a transaction?

**Review by maarcingebala — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Review by michal-macioszczyk — APPROVED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -71,6 +71,7 @@ documentation = "https://docs.saleor.io/"
   Authlib = "^1.0.0"
   pillow-avif-plugin = "^1.3.1"
   semantic-version = "^2.10.0"
+  pytest-memray = "^1.5.0"
```
Comment (NyanKiyoshi) [inline pyproject.toml:74]: This should go onto the development dependencies, not prod ones

**Pull request merged**